File: android/src/main/java/org/axen/flutter/glide/converter/DrawableConverter.java
package org.axen.flutter.glide.converter;

import android.content.Context;

import androidx.annotation.NonNull;

import com.bumptech.glide.RequestBuilder;

import java.lang.reflect.Field;

public final class DrawableConverter<T> implements ImageConverter<T> {
    @NonNull
    public RequestBuilder<T> convert(
            @NonNull Context context,
            @NonNull RequestBuilder<T> builder,
            @NonNull Object resource
    ) throws Exception {
        String[] classGroup = ((String) resource).split("\\.");
        String packageName = context.getPackageName();
        Class clazz = Class.forName(packageName + '.' + classGroup[0] + '$' + classGroup[1]);
        Field field = clazz.getDeclaredField(classGroup[2]);
        Object value = field.get((Object) null);
        return builder.load(value);
    }
}

File: android/src/main/kotlin/org/axen/flutter/glide/converter/DrawableConverter.kt
package org.axen.flutter.glide.converter

import android.content.Context
import com.bumptech.glide.RequestBuilder
import org.axen.flutter.glide.common.ImageConverter

class DrawableConverter<T>: ImageConverter<T> {
    override fun convert(
        context: Context,
        builder: RequestBuilder<T>,
        resource: Any
    ): RequestBuilder<T> {
        val classGroup = (resource as String).split(".")
        val packageName = context.packageName
        val clazz = Class.forName("$packageName.${classGroup[0]}$${classGroup[1]}")
        val field = clazz.getDeclaredField(classGroup[2])
        val value = field.get(null) as Int
        return builder.load(value)
    }

}